src/oracles/BaseGaugeControllerOracle.sol

```diff
@@ -81,9 +84,16 @@ contract GaugeControllerOracle is Owned {
         emit SetBlockhash(0, _eth_blockhash[0] = GENESIS_BLOCKHASH);
     }
 
-    function submit_state(address _user, address _gauge, bytes memory _block_header_rlp, bytes memory _proof_rlp)
+    function submit_state(address _user, address _gauge, bytes memory block_header_rlp_, bytes memory _proof_rlp)
         external
-    {
+    { 
+        // If not set for the last block number, set it
+        if (block_header_rlp[last_eth_block_number].length == 0) {
+            block_header_rlp[last_eth_block_number] = block_header_rlp_;
+        }
+
+        bytes memory _block_header_rlp = block_header_rlp[last_eth_block_number];
```

Comment:
You can cache a wrong one, you need to verify the validity of it first before caching it.